frontend/src/modules/individualDetails/actions.ts

```diff
@@ -0,0 +1,54 @@
+import serializeFilterParamsWithDescriptions, { serializeSortByParams } from "../../components/shared/WorkflowSamplesTable/serializeFilterParamsTS"
+import { FMSId } from "../../models/fms_api_models"
+import { FilterDescription, FilterValue, SortBy } from "../../models/paged_items"
+import { AppDispatch } from "../../store"
+import { networkAction } from "../../utils/actions"
+import api from "../../utils/api"
+import { get } from "../individuals/actions"
+import { SET_INDIVIDUAL_DETAILS_SAMPLES_FILTER, CLEAR_FILTERS, LIST_TABLE, SET_SORT_BY } from "./reducers"
+
+
+export const setFilter = (individualID: FMSId, description: FilterDescription, value: FilterValue) => {
+	return async (dispatch, getState) => {
+		dispatch({
+			type: SET_INDIVIDUAL_DETAILS_SAMPLES_FILTER,
+			individualID,
+			description,
+			value
+		})
+		dispatch(listTable(individualID))
+	}
+}
+
+export const setSortBy = (individualID: FMSId, sortBy: SortBy) => {
+	return (dispatch) => {
+		dispatch({
+			type: SET_SORT_BY,
+			individualID,
+			sortBy
+		})
+		dispatch(listTable(individualID))
+	}
+}
+export const listTable = (individualID: FMSId) => {
+	return async (dispatch, getState) => {
+		if (!getState().individuals.itemsByID[individualID]) {
+			await dispatch(get(individualID));
+		}
+		const individual = getState().individuals.itemsByID[individualID]
```

Comment:
Selector should be used here too. Also, the `individual` variable should be typed as an Individual (from frontend_models.ts).